snp_application/src/snp_widget.cpp

```diff
@@ -1,686 +1,197 @@
+#include <QScrollBar>
+#include <snp_tpp/tpp_widget.h>
+#include <trajectory_preview/trajectory_preview_widget.h>
 
-  auto request = std::make_shared<snp_msgs::srv::GenerateScanMotionPlan::Request>();
-  scan_traj_motion_planning_client_->async_send_request(
-      request, std::bind(&SNPWidget::onScanGenerationDone, this, std::placeholders::_1));
-}
+static const std::string BT_FILES_PARAM = "bt_files";
+static const std::string BT_PARAM = "tree";
+static const std::string BT_SHORT_TIMEOUT_PARAM = "bt_short_timeout";
+static const std::string BT_LONG_TIMEOUT_PARAM = "bt_long_timeout";
 
-void SNPWidget::onScanGenerationDone(rclcpp::Client<snp_msgs::srv::GenerateScanMotionPlan>::SharedFuture future)
+class TPPDialog : public QDialog
 {
-  snp_msgs::srv::GenerateScanMotionPlan::Response::SharedPtr response = future.get();
-  if (response->success)
-  {
-    emit log("Successfully generated scan motion plan");
-    emit updateStatus(true, SCAN_MOTION_PLANNING_ST, SCAN_APPROACH_ST, STATES.at(SCAN_MOTION_PLANNING_ST));
-  }
-  else
+public:
+  TPPDialog(rclcpp::Node::SharedPtr node, QWidget* parent = nullptr) : QDialog(parent)
   {
-    QString message;
-    QTextStream ss(&message);
-    ss << "Failed to generate scan motion plan: '" << QString::fromStdString(response->message) << "'";
-    emit log(message);
-    emit updateStatus(false, SCAN_MOTION_PLANNING_ST, SCAN_MOTION_PLANNING_ST, STATES.at(SCAN_MOTION_PLANNING_ST));
-    return;
-  }
+    setWindowTitle("Tool Path Planner");
 
-  scan_traj_ = response->motion_plan;
-  auto request = std::make_shared<snp_msgs::srv::ExecuteMotionPlan::Request>();
-  request->use_tool = false;
-  request->motion_plan.header = scan_traj_.header;
-  request->motion_plan.joint_names = scan_traj_.joint_names;
-  request->motion_plan.points.push_back(scan_traj_.points.at(0));
-  request->motion_plan.points.push_back(scan_traj_.points.at(1));
+    boost_plugin_loader::PluginLoader loader;
+    loader.search_libraries.insert(NOETHER_GUI_PLUGINS);
+    loader.search_libraries.insert(SNP_TPP_GUI_PLUGINS);
+    loader.search_libraries_env = NOETHER_GUI_PLUGIN_LIBS_ENV;
+    loader.search_paths_env = NOETHER_GUI_PLUGIN_PATHS_ENV;
 
-  auto cb = std::bind(&SNPWidget::onScanApproachDone, this, std::placeholders::_1);
-  motion_execution_client_->async_send_request(request, cb);
+    auto* w = new snp_tpp::TPPWidget(node, std::move(loader), this);
 
-  scan_complete_ = false;
-
-  // Reset any tool paths or motion plans based on a previous scan
-  tool_paths_.clear();
-  motion_plan_.reset();
-}
-
-void SNPWidget::onScanApproachDone(FJTResult result)
-{
-  snp_msgs::srv::ExecuteMotionPlan::Response::SharedPtr response = result.get();
-  if (response->success)
-  {
-    emit updateStatus(true, SCAN_APPROACH_ST, START_RECONSTRUCTION_ST, STATES.at(START_RECONSTRUCTION_ST));
-    emit log("Successfully executed scan approach motion");
+    auto* vl = new QVBoxLayout(this);
+    vl->addWidget(w);
```

Comment:
This might've come from an example somewhere, but variable names `w` and `vl` aren't really descriptive.